### PR TITLE
Fix: Build package prior to installing

### PR DESCRIPTION
**Pull request by ta2edchimp**
The `install` script has been adjusted to invoke `npm run build` on `install` if the `dist` dir is not present (that is, when fetching from git or sth like that).

**Note for Windows**
I have no idea if this solution would suffice in Windows environments. I guess the script will fail due to the default command line (`cmd.exe`) not recognizing the statement. Any ideas?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "report-coverage": "cat ./coverage/lcov.info | codecov",
     "validate": "npm t && npm run check-coverage",
     "commit": "git-cz",
-    "install": "opt --out ghooks-install --exec \"node ./bin/install\"",
+    "install": "opt --out ghooks-install --exec \"[ ! -d \"./dist\" ] && npm run build; node ./bin/install\"",
```
Comment (ta2edchimp) [inline package.json]: When cloning / fetching from git using a shell on Windows, I guess people use Git bash or the like?
Because when running `npm install` on Win's default cmd line prompt `cmd.exe` it will possibly fail due to this new statement :disappointed:

**Comment by gtramontina** [comment]
I think this solution is enough for the time being… Perhaps moving `[ ! -d \"./dist\" ] && npm run build;` to a `preinstall` task (still behind `opt --out ghooks-install`) would separate the concerns a little better?

We can figure windows out later…

**Comment by ta2edchimp** [comment]
I have not yet figured out exactly *when* preinstall will be invoked, _before_ or _after_ installing dependencies dependencies.

**Comment by gtramontina** [comment]
Oh right… we need our dependencies installed in order to run `npm run build`… o_O
Merging! Thanks a lot, @ta2edchimp! :beers:

**Comment by ta2edchimp** [comment]
Hmm ... I had a few quick tests and in fact (at least on `npm@3.8.3`) `preinstall` **is** invoked after dependencies have been installed, directly before `install`.

Although exactly what we'd like to achieve, imho this

1. is neither a perfect nor the correct behaviour (it should be invoked before installing dependencies, to allow hooking into their installations for example) but a discussion about this would be out of this scope here,
2. is also nothing to rely on. Compare [multi-stage install behavior](https://github.com/npm/npm/issues/6926#issuecomment-105260659) — this means we could run into unexpected issues later on:

>The order that preinstall lifecycles are run is not deterministic as they're not supposed to depend on each other.

tl;dr: I too think it's best to stick to the behavior now introduced and just figure out any issues that may arise on Windows.

**Comment by gtramontina** [comment]
…and it failed fast: https://github.com/gtramontina/ghooks/issues/63
Looking into fixing it now!
